javapoet/src/main/java/com/palantir/javapoet/CodeWriter.java

```diff
@@ -418,10 +418,7 @@ String lookupName(ClassName className) {
             return join(".", className.simpleNames());
         }
 
-        // We'll have to use the fully-qualified name. Mark the type as importable for a future pass.
-        if (!javadoc) {
-            importableType(className);
-        }
```

Comment:
One risk is that type references in javadoc may exist outside of a `{@link T}` or `@see T`, where the import will be unused and removed by the formatter, losing precise type information. Should `$T` be used in such scenarios? I'm not really sure.

Reply:
If we want to support importing types in Javadoc, then I think you have to accept this. You need some way to signal that a type should be imported, and `$T` is the (only) sensible way to do that.

Reply:
If this happens, it's a pretty simple fix once the problem is noticed.

Reply:
It's also hopefully something you would notice when switching to this library and your expected test output changes.

Reply:
👍 this is a reasonable tradeoff imo, and makes the results more readable. If we encounter unexpected edge cases, we can always make it configurable similarly to other prefs like whether or not `java.lang` classes are imported.